### PR TITLE
Adding Evidential, Mean-Variance (MVE), and Binary Dirichlet Uncertainty Predictors

### PR DESCRIPTION
**Pull request by akshatzalte**
## Description
This PR implements the MVE, Evidential, and binary classification Dirichlet uncertainty predictors.

Added these to the `UncertaintyPredictor` module.

## Example / Current workflow
```
from chemprop.uncertainty.predictor import EvidentialAleatoricPredictor
predictor = EvidentialAleatoricPredictor()
predss, uncss = predictor(test_loader, models, trainer)

### CODE REVIEW

File: examples/uncertainty.ipynb
Comment (shihchengli) [inline examples/uncertainty.ipynb:1]: I think we can remove this change here. I will open another PR to make this notebook more general and comprehensive.

File: chemprop/uncertainty/predictor.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 from lightning import pytorch as pl
 import torch
 from torch import Tensor
+import torch._tensor
```
Comment (shihchengli) [inline chemprop/uncertainty/predictor.py]:
```suggestion
```

**Review by shihchengli — CHANGES_REQUESTED** [review]

File: chemprop/uncertainty/predictor.py [hunk 2]
```diff
@@ -54,10 +54,20 @@ def __call__(
 
 @UncertaintyPredictorRegistry.register("mve")
 class MVEPredictor(UncertaintyPredictor):
+    """
+    Class that estimates prediction means and variances (MVE).
```
Comment (shihchengli) [inline chemprop/uncertainty/predictor.py:58]:
```suggestion
    Class that estimates prediction means and variances (MVE). [nix1994]_
    
    References
    ----------
    .. [nix1994] Nix, D. A.; Weigend, A. S. "Estimating the mean and variance of the target
        probability distribution." Proceedings of 1994 IEEE International Conference on Neural
        Networks, 1994 https://doi.org/10.1109/icnn.1994.374138
```

**Comment by akshatzalte** [comment]
Notebook changes have been reverted and the binary classification Dirichlet is added.

**Comment by shihchengli** [comment]
Can you do black formatting?

**Review by shihchengli — APPROVED** [review]

**Pull request merged**